src/Hedgehog/Tuple.fs

```diff
@@ -0,0 +1,19 @@
+namespace Hedgehog
+
+module private Tuple =
+
+    let mapFirst (f : 'a -> 'c) (x : 'a, y : 'b) : 'c * 'b =
+        f x, y
+
+    let mapSecond (f : 'b -> 'c) (x : 'a, y : 'b) : 'a * 'c =
```

Comment:
I prefer calling these two functions `map1` and `map2` because then their names are more symmetric at a syntactic level.  I can understand though if others find these names too terse.

Reply:
What's the equivalent naming in Haskell/Hoogle?

Reply:
The F# pair type `'a * 'b` would be an instance of the Haskell type class [Data.Bifunctor](https://hackage.haskell.org/package/bifunctors-5/docs/Data-Bifunctor.html), so the equivalent naming in Haskell would be `first` and `second`, which is neither `mapFirst` and `mapSecond` nor `map1` and `map2`.

I think `first` and `second` are bad names.  It is generally impossible to [get the value out of your ~monad~ _functor_](https://blog.ploeh.dk/2019/02/04/how-to-get-the-value-out-of-the-monad/), but it is possible with the pair type `'a * 'b`, and in this case, I would expect `first` and `second` to have that behavior.  More specifically, I expect their types to be `first : 'a * 'b -> 'a` and `second : 'a * 'b -> 'b`.  In fact, F# has functions with those exact types and abbreviations of those names as `fst` and `snd`.

For that reason, I do not recommend following the Haskell naming convention here.  I am fine with either `mapFirst` and `mapSecond` or `map1` and `map2` but with a preference for the latter.

What about calling them `mapFst` and `mapSnd`?  That satisfies my preference for the names to be my syntactically similar (i.e. they are the same length) while also being consistent with existing functions in the core F# library.

Reply:
I also like `map1` and `map2`. The F# core library can't call them `1` and `2` because they're illegal identifiers, but in context `map1` and `map2` are very readable (c.f. `mapFst`, `mapSnd`).

Reply:
I'm not sure... Perhaps, in this case, `map1` and `map2` clash with function arity(?) ― Also, nothing actually stops us from having those inlined as long as they don't violate the rule of three.

In fact:
*"[..]wait, until you have more facts available to you. You don't have to take the rule literally either. You can wait until you have four, five, or six examples of the duplication, if the rate of change is low."*
-- https://blog.ploeh.dk/2014/08/07/why-dry/

Reply:
> Perhaps, in this case, `map1` and `map2` clash with function arity(?)

That is possible.

> nothing actually stops us from having those inlined

What do you mean by "inlined"?

Reply:
`map1`, `map2` etc for this use conflicts directly with the standard library. See `List.map2`, `List.map3`, where the number is a reference to the arity of the function being mapped.

`mapFst` or `mapSnd` fit in better with the standard library, which provides a `fst` and `snd` function, and also a plethora of `map` functions. Though lately I've taken to providing a sub module to group these things and just keeping the simple name. For example `Result.Error.map` vs `Result.mapError`.

Reply:
Good points.

> `mapFst` or `mapSnd` fit in better with the standard library, which provides a `fst` and `snd` function, and also a plethora of `map` functions. Though lately I've taken to providing a sub module to group these things and just keeping the simple name. For example `Result.Error.map` vs `Result.mapError`.

Oh, yes.  I too have been writing more code like `Result.Error.map`.  I put all the nouns first and then the name of the function gets to be a single verb.

Reply:
Renamed to `mapFst`, `mapSnd`.

Reply:
The PR has already been completed, so this is a bit late.  Nonetheless, another point of comparison is with [FSharpPlus, which calls these `mapItem1` and `mapItem2`.](https://github.com/fsprojects/FSharpPlus/blob/a36d35a3470a0a59b176e7527f426f9014333c1d/src/FSharpPlus/Extensions/Tuple.fs)